src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx

```diff
@@ -253,18 +253,37 @@ export default class GeneralUserSettingsTab extends React.Component<IProps, ISta
         PlatformPeg.get()?.setSpellCheckEnabled(spellCheckEnabled);
     };
 
-    private onPasswordChangeError = (err: { error: string } & MatrixError): void => {
-        // TODO: Figure out a design that doesn't involve replacing the current dialog
-        let errMsg = err.error || err.message || "";
-        if (err.httpStatus === 403) {
-            errMsg = _t("Failed to change password. Is your password correct?");
-        } else if (!errMsg) {
-            errMsg += ` (HTTP status ${err.httpStatus})`;
+    private onPasswordChangeError = (err: Error): void => {
+        logger.error("Failed to change password: " + err);
+
+        let underlyingError = err;
+        if (err instanceof UserFriendlyError && err.cause instanceof Error) {
+            underlyingError = err.cause;
         }
-        logger.error("Failed to change password: " + errMsg);
+
+        const errorMessage = extractErrorMessageFromError(
+            err,
+            _t("Unknown password change error (%(stringifiedError)s)", {
+                stringifiedError: String(err),
+            }),
+        );
+
+        let errorMessageToDisplay;
+        if (underlyingError instanceof HTTPError && underlyingError.httpStatus === 403) {
+            errorMessageToDisplay = _t("Failed to change password. Is your password correct?");
+        } else if (underlyingError instanceof HTTPError) {
+            errorMessageToDisplay = _t("%(errorMessage)s (HTTP status %(httpStatus)s)", {
+                errorMessage,
+                httpStatus: underlyingError.httpStatus,
+            });
+        } else {
+            errorMessageToDisplay = errorMessage;
+        }
+
+        // TODO: Figure out a design that doesn't involve replacing the current dialog
         Modal.createDialog(ErrorDialog, {
```

Comment:
Remaining [strict Typescript failures](https://github.com/matrix-org/matrix-react-sdk/actions/runs/4705715410/jobs/8346457726) are unlrelated to this PR and the problem is generally tracked by https://github.com/vector-im/element-web/issues/24899